### ISSUE TITLE
SetStateAsync in ActorStateManager should not check for type

### ISSUE THREAD

**Issue by samneirinck** [loekd/ServiceFabric.Mocks#25]
When I want to mock the following code in an actor:
```            
await StateManager.SetStateAsync(stateName, string.Empty);
await StateManager.SetStateAsync(stateName, 5);
await StateManager.SetStateAsync(stateName, DateTimeOffset.UtcNow);
```
It fails with an `InvalidCastException` trying to cast 5 to string. It is actually valid code when running under Service Fabric itself.

If you agree, I can create a PR to fix this.

**Comment by loekd** [comment]
I've fixed the problem in v 2.2.4. as the project was open. PR's are always welcome!

**Comment by samneirinck** [comment]
Thanks, didn't want to create a PR yet without confirmation that it was a bug.
